marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/client/MarkLogicClient.java

```diff
@@ -337,7 +337,7 @@ public void sendAdd(Reader in, String baseURI, RDFFormat dataFormat, Resource...
 	 */
 	public void sendAdd(String baseURI, Resource subject, IRI predicate, Value object, Resource... contexts) throws MarkLogicRdf4jException {
 		if (WRITE_CACHE_ENABLED) {
-			timerWriteCache.add(subject, predicate, object, contexts);
+			timerWriteCache.add(subject, predicate, skolemize(object), contexts);
```

Comment:
As we discussed, we do have to skolemize bnodes on ingest.  So subject, predicate, and object must go through skolemization.